### ISSUE TITLE
Bluebird function expressions shouldn't be named

### ISSUE THREAD

**Issue by phpnode** [6to5/6to5#552]
The following code:
```js
async function hello () {
  await hello();
}
```
compiles to:
```js
var hello = _bluebird.coroutine(function* hello() {
  yield hello();
});
```

But this is not correct, the `hello` inside the function refers to the generator when it should refer to the wrapped function. Fixing this is a little tricky, I think you'd need to introduce a new variable and do some renaming:
```js
var hello$tmp = _bluebird.coroutine(function* hello() {
  yield hello$tmp();
});
var hello = hello$tmp;
```

**Comment by sebmck** [comment]
It's probably worth just stripping the `id` since `bluebird.coroutine` wont retain the name anyway.

**Comment by phpnode** [comment]
ah yeah, true

**Comment by sebmck** [comment]
Fixed in the 3.0.0 branch

**Comment by sebmck** [comment]
Fixed as of 3.0.0
